packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart

```diff
@@ -55,6 +57,31 @@ class TimelineScreen extends Screen {
   @override
   String get docPageId => 'timeline';
 
+  @visibleForTesting
+  static const debugModeWarning =
+      DebugModePerformanceMessage(DevToolsScreenType.timeline);
+  @visibleForTesting
+  static const profileGranularityWarning =
+      HighProfileGranularityMessage(DevToolsScreenType.timeline);
+
+  @override
+  List<Widget> messages(BuildContext context) {
+    final showDebugModeWarning = serviceManager.connectedApp.isFlutterAppNow &&
+        !serviceManager.connectedApp.isProfileBuildNow;
+    final timelineController = Controllers.of(context).timeline;
+    final showProfileGranularityWarning = timelineController
+            .cpuProfilerController
+            .service
+            .profileGranularityFlagNotifier
+            .value
+            .valueAsString ==
+        ProfileGranularity.high.value;
+    return [
```

Comment:
this is duplicated code from the Performance  screen. Refactor to add a base class or a helper method.